iocore/cache/CacheWrite.cc

```diff
@@ -617,7 +617,6 @@ Vol::evacuateDocReadDone(int event, Event *e)
   if (doc->magic != DOC_MAGIC) {
     Debug("cache_evac", "DOC magic: %X %d", (int)dir_tag(&doc_evacuator->overwrite_dir),
```

Comment:
@scw00 There is only this debug log between the if and the assert, how the assert help us?

Reply:
Because it is unexpected behavior. It should not happen in normal processing. We should collect more information instead of throw a debug log. 

On the other way, IIUC, the data will be truncated if we goto Ldone.

Reply:
Do you mean coming in this if block is unexpected behavior?

Reply:
Yes.

Reply:
Gotcha! Then, it looks like we should use `ink_abort()`.

Reply:
I think `ink_abort` is critical error which make ATS can not serve anymore. But in this case, it only influence one Object (truncated object), so `ink_assert` is enough .

Reply:
`ink_assert` works with debug build only. I agree with `ink_abort` is too much, but we need some error handling if this is unexpected behavior.

Reply:
All right, I get it, thanks a lot